modin/core/execution/ray/common/utils.py

```diff
@@ -186,6 +141,57 @@ def initialize_ray(
         NPartitions._put(num_cpus)
 
 
+def _object_store_memory() -> Optional[int]:
+    """
+    Get the object store memory we should start Ray with, in bytes.
+
+    Returns
+    -------
+    Optional[int]
+        The object store memory size in bytes.
+    """
+    object_store_memory = Memory.get()
+    if object_store_memory is not None:
+        return object_store_memory
+    # If the user doesn't want a particular object store size, choose an object
+    # store size according to the amount of available memory.
+    virtual_memory = psutil.virtual_memory().total
+    if sys.platform.startswith("linux"):
+        shm_fd = os.open("/dev/shm", os.O_RDONLY)
+        try:
+            shm_stats = os.fstatvfs(shm_fd)
+            system_memory = shm_stats.f_bsize * shm_stats.f_bavail
+            if system_memory / (virtual_memory / 2) < 0.99:
+                warnings.warn(
+                    f"The size of /dev/shm is too small ({system_memory} bytes). The required size "
+                    + f"at least half of RAM ({virtual_memory // 2} bytes). Please, delete files in /dev/shm or "
+                    + "increase size of /dev/shm with --shm-size in Docker. Also, you can set "
+                    + "the required memory size for each Ray worker in bytes to MODIN_MEMORY environment variable."
+                )
+        finally:
+            os.close(shm_fd)
+    else:
+        system_memory = virtual_memory
+    object_store_memory = int(0.6 * system_memory // 1e9 * 1e9)
+    # If the memory pool is smaller than 2GB, just use the default in ray.
+    if object_store_memory == 0:
+        object_store_memory = None
+    else:
+        # Versions of ray with MAC_DEGRADED_PERF_MMAP_SIZE_LIMIT don't allow us
```

Comment:
I wonder if we should put something about different function behaviour on different platforms in the `Notes` section of the docstring... opinions?

Reply:
I tried to clarify the behavior particular to linux and macs in the main docstring body.